pontoon/sync/tasks.py

```diff
@@ -122,9 +128,11 @@ def sync_project_repo(self, project_pk, repo_pk, project_sync_log_pk, now, obsol
     if not no_pull:
         pull_changes(db_project)
 
-    locales = repo.locales
     if locale:
-        locales = [locale]
+        locales = [locale] if locale in repo.locales else []
+    else:
+        locales = repo.locales
+
```

Comment:
We probably don't need this change, because we're already protected:
https://github.com/mozilla/pontoon/blob/master/pontoon/sync/tasks.py#L68